coral/src/services/api.ts

```diff
@@ -187,7 +198,7 @@ function parseResponseBody<T extends SomeObject>(
     if (response.status === 204) {
       return Promise.resolve({} as unknown as T);
     } else {
-      return response.json().then((data) => data as unknown as T);
+      return checkForKlawErrors(response).then((data) => data as unknown as T);
```

Comment:
I would not do it here, but 

```
function handleResponse<TResponse extends SomeObject>(
  response: Response
): Promise<TResponse> {
  return Promise.resolve(response)
    .then(transformHTTPRedirectToLoginTo401)
    .then(transformHTTPRedirectToRootTo204)
    .then(checkStatus)
    .then((response) => parseResponseBody<TResponse>(response))
    .then(<Give function a good name>) <- new check
    .catch(handleHTTPError);
}
```

Reply:
I thought it may be good to avoid two places where we check `const contentType = response.headers.get("content-type");` and do `response.json()` - but it feels a bit better separated that way, too. Now have to get the typing right :D then I'll push the change!